src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -147,16 +153,40 @@ def ensure_output_dirs_exist(self):
 class RscCompile(ZincCompile):
   """Compile Scala and Java code to classfiles using Rsc."""
 
-  _name = 'rsc' # noqa
+  _compiler_lookup = {
+    # NB It is set up this way to simplify the tag names, but it would be more precise to have the
+    # tags use the workflow names directly
+    'rsc': 'rsc-then-zinc',
+    'zinc': 'zinc-only'
+  }
+
+  _name = 'mixed' # noqa
   compiler_name = 'rsc'
 
   @classmethod
   def implementation_version(cls):
     return super(RscCompile, cls).implementation_version() + [('RscCompile', 172)]
 
+  def __init__(self, *args, **kwargs):
+    super(RscCompile, self).__init__(*args, **kwargs)
+
+    self._rsc_compat_tag = self.get_options().force_compiler_tag_prefix
+    self._compiler_tags = {'{}:{}'.format(self._rsc_compat_tag, k): v for k,v in self._compiler_lookup.items()}
+    self._default_workflow = self.get_options().default_workflow
+
+    # If the default workflow is conveyed through a flag, override tag values.
+    default_workflow_rank = self.get_options().get_rank('default_workflow')
```

Comment:
I believe this and the next line could be replaced with:
```suggestion
    default_workflow_rank = self.get_options().is_flagged('default_workflow')
```

Reply:
That has slightly different semantics, but maybe it's better to ignore env vars in this case?

Reply:
Doing it.

Reply:
Oh oops! I didn't realize that, had literally just learned about `.is_flagged()` that day.

Reply:
If we want to support env vars, I would leave a comment saying that is supported even though it's clear from the definition of `CLI_RANKS`, because when my brain thinks "cli flags" it thinks flags provided with hyphens to the right of the `./pants`. If we were to do that, I might also propose adding an `.is_flagged_or_from_env()` method which does this conditional to `OptionValueContainer`.